### PR TITLE
TS Migration

### PR DESCRIPTION
**Pull request by matteoJeulin**

### CODE REVIEW

File: package.json
Comment (ersbeth) [inline package.json:1]: Add the "types" field like in other repositories (I'm not talking about the "type" field which is correct)

File: rollup.config.js [hunk 1]
```diff
@@ -1,51 +1,49 @@
...
-            format: "iife",
-            name: MODULE_NAME,
-            file: `${DIST}/${MODULE_FILENAME}.min.js`,
-            plugins: [
-                terser(), // minify
-            ],
+            sourcemap: true,
         },
     ],
+    // TODO: uncomment this when package is fully ported to TS
```
Comment (ersbeth) [inline rollup.config.js]: Uncomment this as stated

File: rollup.config.js [hunk 2]
```diff
@@ -1,51 +1,49 @@
 import commonjs from "@rollup/plugin-commonjs";
 import { nodeResolve } from "@rollup/plugin-node-resolve";
-import { terser } from "rollup-plugin-terser";
+import typescript from "@rollup/plugin-typescript";
+import del from 'rollup-plugin-delete'
+import { dts } from "rollup-plugin-dts";
+import { readFileSync } from 'fs';
+
 
 const MODULE_NAME = "ImageSkeletonizer";
```
Comment (ersbeth) [inline rollup.config.js]: I think this var is not needed anymore

File: rollup.config.js [hunk 3]
```diff
@@ -1,51 +1,49 @@
...
 const MODULE_FILENAME = "image-skeletonizer";
```
Comment (ersbeth) [inline rollup.config.js:8]: same

File: rollup.config.js [hunk 4]
```diff
@@ -1,51 +1,49 @@
...
 const DIST = "./dist";
```
Comment (ersbeth) [inline rollup.config.js]: same

**Review by ersbeth — CHANGES_REQUESTED** [review]
Partial review, I can't see diffs for source files. I'll seek for another way to finish the review.

File: src/Image/BinaryImage.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  getValue (x: number, y: number): 0 | 1 {
+    const output = this.data[this.getIndex(x, y)];
+    if (output !== 1 && output !== 0)
+      throw '[BinaryImage] getValue : invalid value, data should only have 0s and 1s';
+    return output;
+  };
+
+/**
+ *  @private
+ *  TODO : tolerance is unused, see if it is useful
```
Comment (ersbeth) [inline src/Image/BinaryImage.ts:38]: So did you find it ?

File: src/Image/IntDistanceImage.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+                    dist = Math.min(dist, this.data[index0_1] + c1);
+                    dist = Math.min(dist, this.data[index_11] + c2);
+                    this.data[index] = Math.min(dist, this.data[index10] + c1);
+                }
+            }
+        }
+    };
+
+    /**
+     *  @return {ImageData} A grey scale ImageData to visualize the distances.
```
Comment (ersbeth) [inline src/Image/IntDistanceImage.ts]: remove type

**Review by ersbeth — CHANGES_REQUESTED** [review]

File: src/Image/IntDistanceImage.ts [hunk 7]
```diff
@@ -1,138 +1,141 @@
...
-        throw "IntDistanceImage Error : source must be an instance of BinaryImage";
+export class IntDistanceImage {
+    coeff: number;
+    width: number;
+    height: number;
+    data: number[];
+
+    constructor(c1: number, c2: number, source: BinaryImage, uncolored: number) {
+        if (!(source instanceof BinaryImage)) {
+            throw "IntDistanceImage Error : source must be an instance of BinaryImage";
```
Comment (ersbeth) [inline src/Image/IntDistanceImage.ts]: format error message

File: src/Skeletonizer/CapsuleDistance.ts [hunk 8]
```diff
@@ -1,64 +1,60 @@
...
+        const p1p_l = v.length();
+        const p1p_sqrl = p1p_l * p1p_l;
 
         // In unit_dir basis, vector (this.r1-this.r2, this.length) is normal to the "weight line"
         // We need a projection in this direction up to the segment line to know in which case we fall.
 
-        var x_p_2D = v.x*unit_dir.x + v.y*unit_dir.y;
-        // pythagore inc.
-        var y_p_2D = Math.sqrt(
-            Math.max( // Necessary because of rounded errors, pyth result can be <0 and this causes sqrt to return NaN...
```
Comment (ersbeth) [inline src/Skeletonizer/CapsuleDistance.ts:34]: comment was lost

File: src/Skeletonizer/SkeletonImage.ts [hunk 9]
```diff
@@ -1,397 +1,386 @@
 "use strict";
```
Comment (ersbeth) [inline src/Skeletonizer/SkeletonImage.ts]: Remove all "use strict"  in all files: this is by default when using ESM syntax.

File: src/Skeletonizer/Skeletonizer.ts [hunk 10]
```diff
@@ -1,409 +1,307 @@
...
-const Vector2D = require("./Vector2D");
-
-var Skeletonizer = function(skel_img, dist_img){
-    this.skelImg = skel_img;
-    this.distImg = dist_img;
-};
-
-Skeletonizer.prototype.constructor = Skeletonizer;
-
-/**
```
Comment (ersbeth) [inline src/Skeletonizer/Skeletonizer.ts:14]: Comment was lost

**Review by ersbeth — CHANGES_REQUESTED** [review]

File: src/Image/BinaryImage.ts (hunk 5 above)
Comment (matteoJeulin) [reply src/Image/BinaryImage.ts:38]: It is used when calling the function but is useless in the function itself

**Review by ersbeth — COMMENTED** [review]
Last changes look good to me.

**Review by ersbeth — COMMENTED** [review]
Changes look good to me

**Review by ersbeth — APPROVED** [review]

**Pull request merged**